facebook/react-native
# Use node package dependency to manage JSC version

## Summary

In origin approach, we packed libjsc.so inside react-native.aar and it is difficult for user to choose different JSC variants. E.g., [the Intl supported version](https://github.com/react-native-community/jsc-android-buildscripts#international-variant).

This change list allows application to determine JSC versions or variants by npm/yarn package.

### To use Intl JSC
There is a |useIntlJsc| flag in build.gradle, it will use the same JSC version but with Intl support.

### To use latest JSC

`yarn add jsc-android@canary`

## Changelog

[Android] [Changed] - Allow application to select different JSC variants

**MIGRATION**
Note that there are some changes in build.gradle.
Existing application needs to change their android/build.gradle and android/app/build.gradle.
Hopefully, the rn-diff-purge should handle the case well.

## Test Plan

1. Test RNTester build and run successfully.
2. Create a new project by `react-native init Foo --version /somewhere/react-native-1000.0.0.tgz` and test that build and run successfully.
3. Turn on |useIntlJsc| and add `console.log('Intl type is ' + typeof Intl)` to App.js, verify if the adb log show "Intl type is object"
4. `yarn add jsc-android@canary` and rebuild. Verify adb log `adb logcat | grep 'JavaScriptCore.Version'` and it should return 241213.0.0 not 236355.1.0
